File: Data Structure/Array/Array Data Structure Guide/Basic Problems on Array/Largest element in an Array/method2.java
import java.util.Arrays;

class Main {
    static int findMax(int[] arr, int i) {
      
        // Last index returns the element
        if (i == arr.length - 1) {
            return arr[i];
        }

        // Find the maximum from the rest of the array
        int recMax = findMax(arr, i + 1);

        // Compare with i-th element and return
        return Math.max(recMax, arr[i]);
    }

    static int largest(int[] arr) {
        return findMax(arr, 0);
    }

    public static void main(String[] args) {
        int[] arr = {10, 324, 45, 90, 9808};
        System.out.println(largest(arr));
    }
}

File: Data Structure/Array/Array Data Structure Guide/Basic Problems on Array/Largest element in an Array/method2.cpp
#include <iostream>
#include <vector>
using namespace std;

int findMax(vector<int>& arr, int i) {
  
   // Last index returns the element
    if (i == arr.size() - 1) {
        return arr[i];
    }

    // Find the maximum from the rest of the vector
    int recMax = findMax(arr, i + 1);

    // Compare with i-th element and return
    return max(recMax, arr[i]);
}

int largest(vector<int>& arr) {
  return findMax(arr,0);
}

int main() {
    vector<int> arr = {10, 324, 45, 90, 9808};
    cout << largest(arr);
    return 0;
}